packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -491,6 +491,9 @@ export default createRule<Options, MessageId>({
           node,
           messageId: 'literalBooleanExpression',
           data: {
+            left: valueOfLiteralTypeToExpression(leftStaticValue.value),
```

Comment:
I also investigated using `checker.typeToString()`. The only difference I observed was that that message for enums would look like 
`Unnecessary conditional, comparison is always true, since Foo.a === Foo.a is true` 
rather than 
`Unnecessary conditional, comparison is always true, since 1 === 1 is true`.

Not sure which is better. 🤷‍♂️ Slightly leaning towards the `checker.typeToString()` tbh.

Reply:
🤷 No preference from me, I could see either being nice.